### PR TITLE
[zh-CN]: create  translation for `Set.prototype.difference`

### PR DESCRIPTION
**Pull request by custs21113**
### Description

Translation for `Set.prototype.difference`.

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- 如果 `this` 的元素数量大于 `other.size`，则通过调用 `keys()` 方法遍历 `other`，并使用 `this` 中所有在 `other` 中未见过的元素构造一个新的集合。
+- 否则，它会遍历 `this` 中的元素，并使用 `this` 中的所有使 `other.has(e)` 返回 [假值](/zh-CN/docs/Glossary/Falsy) 的元素 `e` 构造一个新集合。
+
+返回的集合的元素的顺序与 `this` 相同。
+
+## 示例
+
+### 使用 difference() 方法
+
+下面计算奇数集 (<10) 和完全平方集 (<10) 之间的差的示例。其结果是一组不是完全平方数的奇数。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md:46]:
```suggestion
下面计算奇数集（<10）和完全平方集（<10）之间的差的示例。其结果是一组不是完全平方数的奇数。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+{{Specifications}}
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [Polyfill of `Set.prototype.difference` in `core-js`](https://github.com/zloirock/core-js#new-set-methods)
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md:64]:
```suggestion
- [`core-js` 中 `Set.prototype.difference` 的 Polyfill](https://github.com/zloirock/core-js#new-set-methods)
```

File: files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+<math display="block"><semantics><mrow><mi>A</mi><mo>∖</mo><mi>B</mi><mo>=</mo><mo stretchy="false">{</mo><mi>x</mi><mo>∊</mo><mi>A</mi><mo>∣</mo><mi>x</mi><mo>∉</mo><mi>B</mi><mo stretchy="false">}</mo></mrow><annotation encoding="TeX">A\setminus B = \{x\in A\mid x\notin B\}</annotation></semantics></math>
+
+使用维恩图表示：
+
+![两个圆重叠的维恩图。A 不与 B 重叠的区域是 A 和 B 的不同之处。](diagram.svg)
+
+`difference()` 接受 [类集合](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set#set-like_objects) 对象作为 `other` 参数。方法要求 {{jsxref("Operators/this", "this")}} 是一个 {{jsxref("Set")}} 实例，因为它直接检索存储在其中的底层数据，而不调用任何用户代码。然后，它的行为取决于 `this` 和 `other`的大小：
+
+- 如果 `this` 的元素数量大于 `other.size`，则通过调用 `keys()` 方法遍历 `other`，并使用 `this` 中所有在 `other` 中未见过的元素构造一个新的集合。
+- 否则，它会遍历 `this` 中的元素，并使用 `this` 中的所有使 `other.has(e)` 返回 [假值](/zh-CN/docs/Glossary/Falsy) 的元素 `e` 构造一个新集合。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md:38]:
```suggestion
- 否则，它会遍历 `this` 中的元素，并使用 `this` 中的所有使 `other.has(e)` 返回[假值](/zh-CN/docs/Glossary/Falsy)的元素 `e` 构造一个新集合。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+使用数学记号，*差集*的定义如下：
+
+<math display="block"><semantics><mrow><mi>A</mi><mo>∖</mo><mi>B</mi><mo>=</mo><mo stretchy="false">{</mo><mi>x</mi><mo>∊</mo><mi>A</mi><mo>∣</mo><mi>x</mi><mo>∉</mo><mi>B</mi><mo stretchy="false">}</mo></mrow><annotation encoding="TeX">A\setminus B = \{x\in A\mid x\notin B\}</annotation></semantics></math>
+
+使用维恩图表示：
+
+![两个圆重叠的维恩图。A 不与 B 重叠的区域是 A 和 B 的不同之处。](diagram.svg)
+
+`difference()` 接受 [类集合](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set#set-like_objects) 对象作为 `other` 参数。方法要求 {{jsxref("Operators/this", "this")}} 是一个 {{jsxref("Set")}} 实例，因为它直接检索存储在其中的底层数据，而不调用任何用户代码。然后，它的行为取决于 `this` 和 `other`的大小：
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/set/difference/index.md:35]:
```suggestion
`difference()` 接受[类集合](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set#set-like_objects)对象作为 `other` 参数。方法要求 {{jsxref("Operators/this", "this")}} 是一个 {{jsxref("Set")}} 实例，因为它直接检索存储在其中的底层数据，而不调用任何用户代码。然后，它的行为取决于 `this` 和 `other` 的大小：
```

**Review by yin1999 — APPROVED** [review]
Thank you @custs21113

**Pull request merged**